### PR TITLE
feat(cli): Add `meltano config --unsafe` flag

### PR DESCRIPTION
**Pull request by ReubenFrankel**
Essentially bypasses behaviour changes made in #7878 to expose sensitive setting values with `--unsafe`, as per [this request on Slack](https://meltano.slack.com/archives/CKHP6G5V4/p1690300061219669) by @visch.

### CODE REVIEW

**Comment by ReubenFrankel** [comment]
Didn't know whether to keep the yellow styling for sensitive values when they are exposed (as a kind of reminder that the value is still sensitive) or not, so I just made it output what it otherwise would have before #7878. Pretty easy to change if anyone feels strongly.

File: src/meltano/cli/config.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def get_label(metadata) -> str:
     default="json",
 )
 @click.option("--extras", is_flag=True, help="View or list only plugin extras.")
+@click.option("--unsafe", is_flag=True, help="Expose values for sensitive settings.")
```
Comment (edgarrmondragon) [inline src/meltano/cli/config.py:134]: Wdyt about this
```suggestion
@click.option(
    "--safe/--unsafe",
    default=True,
    help="Expose values for sensitive settings.",
)
```

Then you can change instances of `not unsafe` to safe.

Comment (ReubenFrankel) [reply src/meltano/cli/config.py]: 100% agree - changed in d4b7e59.

**Review by edgarrmondragon — APPROVED** [review]
LGTM!

File: src/meltano/cli/config.py [hunk 2]
```diff
@@ -131,6 +131,12 @@ def get_label(metadata) -> str:
     default="json",
 )
 @click.option("--extras", is_flag=True, help="View or list only plugin extras.")
+@click.option(
+    "--safe/--unsafe",
```
Comment (tayloramurphy) [inline src/meltano/cli/config.py:135]: Love this! Can we get a quick update to https://docs.meltano.com/reference/command-line-interface#config as well?

**Review by tayloramurphy — APPROVED** [review]

File: src/meltano/cli/config.py (hunk 2 above)
Comment (edgarrmondragon) [reply src/meltano/cli/config.py:135]: Docs look great. Thanks @ReubenFrankel!

**Pull request merged**